api/session.go

```diff
@@ -0,0 +1,103 @@
+package api
+
+import (
+	"encoding/xml"
+	"errors"
+)
+
+// SessionStatusByReferenceResponse defines the response of the session status
+type SessionStatusByReferenceResponse struct {
+	ProcessingCode        int    `json:"processingCode"`
+	ProcessingDescription string `json:"processingDescription"`
+	ReferenceNumber       string `json:"referenceNumber"`
+	Timestamp             string `json:"timestamp"`
+	Upo                   string `json:"upo"`
+}
+
+// SessionStatusResponse defines the response of the session status
+type SessionStatusResponse struct {
+	Timestamp             string `json:"timestamp"`
+	ReferenceNumber       string `json:"referenceNumber"`
+	ProcessingCode        int    `json:"processingCode"`
+	ProcessingDescription string `json:"processingDescription"`
+	NumberOfElements      int    `json:"numberOfElements"`
+	PageSize              int    `json:"pageSize"`
+	PageOffset            int    `json:"pageOffset"`
+	InvoiceStatusList     []struct {
+	} `json:"invoiceStatusList"`
+}
+
+// TerminateSessionResponse defines the response of the session termination
+type TerminateSessionResponse struct {
+	Timestamp             string `json:"timestamp"`
+	ReferenceNumber       string `json:"referenceNumber"`
+	ProcessingCode        int    `json:"processingCode"`
+	ProcessingDescription string `json:"processingDescription"`
+}
+
+// TerminateSession ends the current session
+func (s *Session) TerminateSession() (*TerminateSessionResponse, error) {
+	response := &TerminateSessionResponse{}
+	var errorResponse ErrorResponse
+	resp, err := s.client.R().
+		SetResult(response).
+		SetError(&errorResponse).
+		SetHeader("SessionToken", s.Token.SessionToken.Token).
+		Get(s.url + "/api/online/Session/Terminate")
+
+	if err != nil {
+		return nil, err
+	}
+	if resp.IsError() {
+		return nil, errors.New(errorResponse.Exception.ExceptionDetailList[0].ExceptionDescription)
+	}
+
+	return response, nil
+}
+
+// GetSessionStatus gets the session status of the current session
+func (s *Session) GetSessionStatus() (*SessionStatusResponse, error) {
```

Comment:
Rather than `Get`, I find `Fetch` more appropriate when there is some kind of network action, leaving `Get` for use-cases when the data can be fetched from memory.
Also, we need to be supplying the context here so that we can get some flow control. (`context.Context` is another thing unique to Go that takes a bit of getting used to.)